### PR TITLE
Switch to poetry for notebook scaffolding

### PR DESCRIPTION
**Pull request by mavam**
This PR switches to a poetry-based workflow for managing the build of example notebooks.

### :memo: Reviewer Checklist

- [ ] Executing `make` in `examples/notebooks/` works
- [ ] Executing `yarn && yarn start` in `web/` executes the notebook workflow

### CODE REVIEW

**Comment by mavam** [comment]
@dispanser since you struggled with building the docs locally, does this branch work for you?

File: examples/notebooks/shell.nix [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# brings poetry into scope and makes sure that poetry-installed C-lib basedpython
```
Comment (dispanser) [inline examples/notebooks/shell.nix:1]: @tobim : maybe you want to have a brief look at this rudimentary `shell.nix`? I know it's not pretty, but at least allows me to run the docs locally.

File: examples/notebooks/shell.nix [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# python dependencies can "see" libstdc++. Ref https://nixos.wiki/wiki/Packaging/Quirks_and_Caveats#ImportError:_libstdc.2B.2B.so.6:_cannot_open_shared_object_file:_No_such_file
+{ pkgs ? (import <nixpkgs> {}).pkgs }:
+with pkgs;
+mkShell {
+  buildInputs = [
+    poetry
```
Comment (dispanser) [inline examples/notebooks/shell.nix:7]: we could add yarn here

**Review by dispanser — APPROVED** [review]
I tested it and got it to run with my NixOS-based setup after some fiddling.

I can now run the docs again locally, which is awesome❗

File: examples/notebooks/shell.nix (hunk 1 above)
Comment (tobim) [reply examples/notebooks/shell.nix:1]: The essential thing is that it does the job. We can worry about "beauty" some other time.

**Review by tobim — APPROVED** [review]

File: examples/notebooks/shell.nix (hunk 2 above)
Comment (mavam) [reply examples/notebooks/shell.nix:7]: That's technically in `web/` and not `examples/notebooks/`, no?

(Resolving, as we are agreeing to ship this.)

**Pull request merged**